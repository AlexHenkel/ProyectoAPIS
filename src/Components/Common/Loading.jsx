import React from 'react'
import styled from 'styled-components'
import { LinearProgress } from 'material-ui/Progress'

const Container = styled.div`
  width: 100%;
  margin-top: 30px;
`

export default () => (
  <Container>
    <LinearProgress color="accent" />
  </Container>
)
