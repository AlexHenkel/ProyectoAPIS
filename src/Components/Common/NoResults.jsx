import React from 'react'
import styled from 'styled-components'
import { Typography as OriginalTypography } from 'material-ui'

const Typography = styled(OriginalTypography)`
  margin-top: 50px !important;
  margin-bottom: 50px !important;
`

export default () => (
  <Typography type="subheading" color="primary" gutterBottom align="center">
    No se encontraron resultados <span>:(</span>
  </Typography>
)
