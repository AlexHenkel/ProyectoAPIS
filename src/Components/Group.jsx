import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { CardContent, CardActions, Button } from 'material-ui'
import { withTheme } from 'material-ui/styles'
import { DisabledButton, Card, CardText } from './Common/Utils'
import ContextMenu, { ContextContainer } from './Common/ContextMenu'
import GroupsActions from '../Data/Redux/GroupsRedux'

const Group = ({ id, allowEdit, onEdit, onRemove, activeGroup, theme,
  name, activeExams, selectActiveGroup }) => (
    <Card key={id} active={activeGroup === id ? 'yes' : ''} theme={theme}>
      <CardContent>
        {allowEdit && (
          <ContextContainer>
            <ContextMenu
              value={id}
              handleEdit={onEdit}
              handleRemove={onRemove}
            />
          </ContextContainer>
        )}
        <CardText type="headline" component="h2" gutterBottom active={activeGroup === id ? 'yes' : ''}>
          {name}
        </CardText>
        <CardText type="body1" gutterBottom active={activeGroup === id ? 'yes' : ''}>
          Quizes activos: <b>{activeExams}</b>
        </CardText>
      </CardContent>
      <CardActions>
        {activeGroup === id && (
          <DisabledButton disabled>Seleccionado</DisabledButton>
        )}
        {activeGroup !== id && (
          <Button dense color="accent" onClick={() => selectActiveGroup(id)}>Ver MÃ¡s</Button>
        )}
      </CardActions>
    </Card>
)

Group.propTypes = {
  id: PropTypes.number.isRequired,
  onEdit: PropTypes.func,
  onRemove: PropTypes.func,
  activeGroup: PropTypes.number.isRequired,
  theme: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  activeExams: PropTypes.number.isRequired,
  selectActiveGroup: PropTypes.func.isRequired,
  allowEdit: PropTypes.bool.isRequired,
}

Group.defaultProps = {
  onEdit: () => null,
  onRemove: () => null,
}

const mapStateToProps = state => ({
  activeGroup: state.groups.activeGroup,
})

const mapDispatchToProps = dispatch => ({
  selectActiveGroup: (id, name) => dispatch(GroupsActions.selectActiveGroup(id, name)),
})

export default connect(mapStateToProps, mapDispatchToProps)(withTheme()(Group))
