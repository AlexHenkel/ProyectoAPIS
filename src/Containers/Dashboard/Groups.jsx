import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Typography,
  CardContent,
  CardActions,
  Button,
} from 'material-ui'
import { withTheme } from 'material-ui/styles'
import AddIcon from 'material-ui-icons/Add'
import { AlignCenter, DisabledButton, Card, CardText } from '../../Components/Utils'
import ContextMenu, { ContextContainer } from '../../Components/ContextMenu'
import ModalSave from '../../Components/ModalSave'
import ModalRemove from '../../Components/ModalRemove'

class Groups extends Component {
  constructor(props) {
    super(props)
    this.state = {
      modalType: 'create',
      modalSaveOpened: false,
      modalRemoveOpened: false,
      toModifyId: -1,
    }
  }

  // /** Fired on add item. Set type of modal and then open modal */
  onAdd = () => this.setState({
    modalType: 'create',
  }, () => this.setState({ modalSaveOpened: true }))

  // /** Fired on update item. Set type of modal, id and then open modal */
  onEdit = id => this.setState({
    modalType: 'update',
    toModifyId: id,
  }, () => this.setState({ modalSaveOpened: true }))

  // /** Function fired on remove item */
  onRemove = id => this.setState({
    toModifyId: id,
    modalRemoveOpened: true,
  })

  onCloseModalSave = () => {
    this.setState({ modalSaveOpened: false })
  }

  onCloseModalRemove = () => {
    this.setState({ modalRemoveOpened: false })
  }

  render() {
    const {
      theme,
      groups,
      selectActiveGroup,
      activeGroup,
    } = this.props
    const {
      modalSaveOpened,
      modalType,
      toModifyId,
      modalRemoveOpened,
    } = this.state
    return (
      <div>
        <Typography type="display1" gutterBottom>Mis grupos</Typography>
        <AlignCenter>
          <Button color="accent" onClick={this.onAdd}>
            <AddIcon />
            Agregar grupo
          </Button>
        </AlignCenter>
        {groups.map(({ id, name, activeQuizes }) => (
          <Card key={id} active={activeGroup === id ? 'yes' : ''} theme={theme}>
            <CardContent>
              <ContextContainer>
                <ContextMenu
                  value={id}
                  handleEdit={this.onEdit}
                  handleRemove={this.onRemove}
                />
              </ContextContainer>
              <CardText type="headline" component="h2" gutterBottom active={activeGroup === id ? 'yes' : ''}>
                {name}
              </CardText>
              <CardText type="body1" gutterBottom active={activeGroup === id ? 'yes' : ''}>
                Quizes activos: <b>{activeQuizes}</b>
              </CardText>
            </CardContent>
            <CardActions>
              {activeGroup === id && (
                <DisabledButton disabled>Seleccionado</DisabledButton>
              )}
              {activeGroup !== id && (
                <Button dense color="accent" onClick={selectActiveGroup(id)}>Ver Más</Button>
              )}
            </CardActions>
          </Card>
        ))}
        <ModalSave
          open={modalSaveOpened}
          title="Grupo"
          onRequestClose={this.onCloseModalSave}
          modalType={modalType}
          toUpdateId={toModifyId}
          fields={[
            {
              type: 'textField',
              id: 1,
              name: 'name',
              label: 'Nombre del grupo',
              required: true,
            },
            {
              type: 'textField',
              id: 2,
              name: 'startDate',
              inputType: 'datetime-local',
              label: 'Fecha de inicio',
              required: true,
            },
            {
              type: 'textField',
              id: 3,
              name: 'endDate',
              inputType: 'datetime-local',
              label: 'Fecha de fin',
              required: true,
            },
          ]}
        />
        <ModalRemove
          open={modalRemoveOpened}
          onRequestClose={this.onCloseModalRemove}
          toRemoveId={toModifyId}
        />
      </div>
    )
  }
}

Groups.propTypes = {
  theme: PropTypes.object.isRequired,
  groups: PropTypes.array,
  activeGroup: PropTypes.number,
  selectActiveGroup: PropTypes.func,
}

Groups.defaultProps = {
  groups: [
    {
      id: 1,
      name: 'Física 1 - ENE/DIC 17',
      activeQuizes: 1,
    },
    {
      id: 2,
      name: 'Física 2 - ENE/DIC 17',
      activeQuizes: 1,
    },
  ],
  activeGroup: 1,
  selectActiveGroup: id => console.log('change group', id),
}

export default withTheme()(Groups)
