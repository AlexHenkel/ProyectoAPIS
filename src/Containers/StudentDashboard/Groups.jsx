import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Typography } from 'material-ui'
import Loading from '../../Components/Common/Loading'
import Group from '../../Components/Group'
import GroupsActions from '../../Data/Redux/GroupsRedux'

class Groups extends Component {
  componentWillReceiveProps({ loading, groups }) {
    const { loading: currLoading, selectActiveGroup } = this.props
    if (!loading && currLoading) {
      if (groups.length) {
        selectActiveGroup(groups[0].id, groups[0].name)
      }
    }
  }

  render() {
    const { loading, groups } = this.props
    return (
      <div>
        <Typography type="display1" gutterBottom>Mis grupos</Typography>
        {loading && <Loading />}
        {!loading && groups.map((group, index) => (
          <Group
            key={index}
            allowEdit={false}
            {...group}
          />
        ))}
      </div>
    )
  }
}

Groups.propTypes = {
  loading: PropTypes.bool.isRequired,
  groups: PropTypes.array.isRequired,
  selectActiveGroup: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  loading: state.groups.get.fetching,
  groups: state.groups.get.results,
})

const mapDispatchToProps = dispatch => ({
  selectActiveGroup: (id, name) => dispatch(GroupsActions.selectActiveGroup(id, name)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Groups)
