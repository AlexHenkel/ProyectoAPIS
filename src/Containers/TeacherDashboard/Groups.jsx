import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Typography, Button } from 'material-ui'
import AddIcon from 'material-ui-icons/Add'
import { AlignCenter } from '../../Components/Common/Utils'
import ModalSave from '../../Components/Common/ModalSave'
import ModalRemove from '../../Components/Common/ModalRemove'
import Loading from '../../Components/Common/Loading'
import Group from '../../Components/Group'
import GroupsActions from '../../Data/Redux/GroupsRedux'

class Groups extends Component {
  constructor(props) {
    super(props)
    this.state = {
      modalType: 'create',
      modalSaveOpened: false,
      modalRemoveOpened: false,
      toModifyId: -1,
    }
  }

  componentWillReceiveProps({ loading, groups }) {
    const { loading: currLoading, selectActiveGroup } = this.props
    if (!loading && currLoading) {
      if (groups.length) {
        selectActiveGroup(groups[0].id, groups[0].name)
      }
    }
  }

  // /** Fired on add item. Set type of modal and then open modal */
  onAdd = () => this.setState({
    modalType: 'create',
  }, () => this.setState({ modalSaveOpened: true }))

  // /** Fired on update item. Set type of modal, id and then open modal */
  onEdit = id => this.setState({
    modalType: 'update',
    toModifyId: id,
  }, () => this.setState({ modalSaveOpened: true }))

  // /** Function fired on remove item */
  onRemove = id => this.setState({
    toModifyId: id,
    modalRemoveOpened: true,
  })

  onCloseModalSave = () => {
    this.setState({ modalSaveOpened: false })
  }

  onCloseModalRemove = () => {
    this.setState({ modalRemoveOpened: false })
  }

  render() {
    const { loading, groups } = this.props
    const { modalSaveOpened, modalType, toModifyId, modalRemoveOpened } = this.state
    return (
      <div>
        <Typography type="display1" gutterBottom>Mis grupos</Typography>
        {loading && <Loading />}
        {!loading && (
          <div>
            <AlignCenter>
              <Button color="accent" onClick={this.onAdd}>
                <AddIcon />
                Agregar grupo
              </Button>
            </AlignCenter>
            {groups.map(group => (
              <Group
                onEdit={this.onEdit}
                onRemove={this.onRemove}
                allowEdit
                {...group}
              />
            ))}
          </div>
        )}
        <ModalSave
          open={modalSaveOpened}
          title="Grupo"
          onRequestClose={this.onCloseModalSave}
          modalType={modalType}
          toUpdateId={toModifyId}
          statePath="groups"
          typePrefix="GROUPS"
          getOnePath="groups.get.results"
          fields={[
            {
              type: 'textField',
              id: 1,
              name: 'name',
              path: 'name',
              label: 'Nombre del grupo',
              required: true,
            },
            {
              type: 'textField',
              id: 2,
              name: 'startDate',
              path: 'startDate',
              inputType: 'datetime-local',
              label: 'Fecha de inicio',
              required: true,
            },
            {
              type: 'textField',
              id: 3,
              name: 'endDate',
              path: 'endDate',
              inputType: 'datetime-local',
              label: 'Fecha de fin',
              required: true,
            },
            {
              type: 'textField',
              id: 4,
              name: 'teacher_id',
              inputType: 'hidden',
              value: 1, // TODO: Change this for user id
              specificFor: 'create',
            },
          ]}
        />
        <ModalRemove
          open={modalRemoveOpened}
          onRequestClose={this.onCloseModalRemove}
          toRemoveId={toModifyId}
          statePath="groups"
          typePrefix="GROUPS"
        />
      </div>
    )
  }
}

Groups.propTypes = {
  theme: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  groups: PropTypes.array.isRequired,
  selectActiveGroup: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  loading: state.groups.get.fetching,
  groups: state.groups.get.results,
})

const mapDispatchToProps = dispatch => ({
  selectActiveGroup: (id, name) => dispatch(GroupsActions.selectActiveGroup(id, name)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Groups)
