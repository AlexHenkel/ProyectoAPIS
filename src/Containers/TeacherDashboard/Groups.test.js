import React from 'react'
import { Groups } from './Groups'
import Loading from '../../Components/Common/Loading'
import Group from '../../Components/Group'
import NoResults from '../../Components/Common/NoResults'

describe('Groups component', () => {
  const groupsRender = ({
    loading = false,
    groups = [],
    selectActiveGroup = jest.fn(),
    userId = 1,
  }) => shallow(<Groups
    loading={loading}
    groups={groups}
    selectActiveGroup={selectActiveGroup}
    userId={userId}
  />)

  it('renders correctly', () => {
    const tree = groupsRender({})
    expect(tree).toMatchSnapshot()
  })

  it('loading bar shown on loading', () => {
    const tree = groupsRender({ loading: true })
    expect(tree.find(Loading).length).toBe(1)
  })

  it('loading bar hide on not loading', () => {
    const tree = groupsRender({ loading: false })
    expect(tree.find(Loading).length).toBe(0)
  })

  it('groups not showing when loading', () => {
    const tree = groupsRender({ loading: true, groups: [1, 2, 3] })
    expect(tree.find(Loading).length).toBe(1)
    expect(tree.find(Group).length).toBe(0)
  })

  it('group results shown correctly', () => {
    const tree = groupsRender({ loading: false, groups: [1, 2, 3] })
    expect(tree.find(Loading).length).toBe(0)
    expect(tree.find(Group).length).toBe(3)
  })

  it('not results message on empty results', () => {
    const tree = groupsRender({ loading: false, groups: [] })
    expect(tree.find(Loading).length).toBe(0)
    expect(tree.find(Group).length).toBe(0)
    expect(tree.find(NoResults).length).toBe(1)
  })
})
