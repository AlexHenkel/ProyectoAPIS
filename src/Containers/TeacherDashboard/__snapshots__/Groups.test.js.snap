// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups component renders correctly 1`] = `
<div>
  <withStyles(Typography)
    gutterBottom={true}
    type="display1"
  >
    Mis grupos
  </withStyles(Typography)>
  <div>
    <styled.div>
      <withStyles(Button)
        color="accent"
        onClick={[Function]}
      >
        <pure(Add) />
        Agregar grupo
      </withStyles(Button)>
    </styled.div>
    <Component />
  </div>
  <Connect(ModalSave)
    fields={
      Array [
        Object {
          "id": 1,
          "label": "Nombre del grupo",
          "name": "name",
          "path": "name",
          "required": true,
          "type": "textField",
        },
        Object {
          "help": "A partir de esta fecha, el quiz estará disponible para tí y tus alumnos",
          "id": 2,
          "label": "Fecha de inicio",
          "name": "startDate",
          "path": "startDate",
          "required": true,
          "type": "datepicker",
        },
        Object {
          "help": "Después de esta fecha, el grupo se cerrará automáticamente",
          "id": 3,
          "inputType": "datetime-local",
          "label": "Fecha de fin",
          "name": "endDate",
          "path": "endDate",
          "required": true,
          "type": "datepicker",
        },
        Object {
          "id": 4,
          "inputType": "hidden",
          "name": "teacherId",
          "specificFor": "create",
          "type": "textField",
          "value": 1,
        },
      ]
    }
    getOnePath="groups.get.results"
    modalType="create"
    onRequestClose={[Function]}
    open={false}
    statePath="groups"
    title="Grupo"
    toUpdateId={-1}
    typePrefix="GROUPS"
  />
  <Connect(ModalRemove)
    onRequestClose={[Function]}
    open={false}
    statePath="groups"
    toRemoveId={-1}
    typePrefix="GROUPS"
  />
</div>
`;
